view/src/storage.rs

```diff
@@ -756,6 +870,9 @@ impl Storage {
             let position = (u64::from(note_record.position)) as i64;
             let source = note_record.source.to_bytes().to_vec();
 
+            // We might have already seen the notes in the form of advice,
+            // so we use ON CONFLICT DO NOTHING to skip re-inserting them
+            // in that case.
```

Comment:
Since we now use the advice mechanism to pre-record notes, it's no longer the case that note insertions happen exactly once: we'll insert the note in the `notes` table when we get it as advice, and then attempt to insert again when processing a `SpendableNoteRecord` when we see it in scanning.  We just have to ignore the second insertion; we know the data can't be different because the note commitment is a hash of the entire note contents, so if the note commitment matches, so does the rest of the row.